### PR TITLE
RAC-1028: Job execution sort [front]

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

Make job execution table columns sortable.

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx [hunk 1]
```diff
@@ -1,39 +1,40 @@
 import React from 'react';
 import {Table} from 'akeneo-design-system';
 import {useDateFormatter, useTranslate} from '@akeneo-pim-community/shared';
-import {JobExecutionRow} from '../../models/JobExecutionTable';
+import {JobExecutionRow} from '../../models';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:4]:
```suggestion
import {JobExecutionRow, JobExecutionFilterSort} from '../../models';
```

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx [hunk 2]
```diff
@@ -1,39 +1,40 @@
...
 import {JobExecutionStatus} from '../JobExecutionStatus';
+import {JobExecutionFilterSort} from '../../models';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:6]:
```suggestion
```

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/hooks/useJobExecutionTable.ts [hunk 3]
```diff
@@ -13,6 +13,8 @@ const useJobExecutionTable = ({page, size, type, status}: JobExecutionFilter): J
         body: JSON.stringify({
           page: page.toString(),
           size: size.toString(),
+          sort_column: sort.column,
+          sort_direction: sort.direction,
```
Comment (juliensnz) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/hooks/useJobExecutionTable.ts:17]:
```suggestion
          sort,
```

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx [hunk 4]
```diff
@@ -1,39 +1,40 @@
...
   jobExecutionRows: JobExecutionRow[];
+  onSortChange: (sort: JobExecutionFilterSort) => void;
+  currentSort: JobExecutionFilterSort;
 };
 
-const JobExecutionTable = ({sticky, jobExecutionRows}: JobExecutionTableProps) => {
+const JobExecutionTable = ({sticky, jobExecutionRows, onSortChange, currentSort}: JobExecutionTableProps) => {
   const translate = useTranslate();
   const dateFormatter = useDateFormatter();
+  const columnHeaders = ['job_name', 'type', 'started_at', 'username', 'status', 'warning_count'];
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:18]: Maybe extract this outside the component

**Review by juliensnz — APPROVED** [review]

File: src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx [hunk 5]
```diff
@@ -1,39 +1,40 @@
...
-        <Table.HeaderCell>
-          {translate('akeneo_job_process_tracker.job_execution_list.table.headers.status')}
-        </Table.HeaderCell>
-        <Table.HeaderCell>
-          {translate('akeneo_job_process_tracker.job_execution_list.table.headers.warnings')}
-        </Table.HeaderCell>
+        {columnHeaders.map(columnHeader => (
+          <Table.HeaderCell
+            key={columnHeader}
+            isSortable
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/front/process-tracker/src/feature/components/JobExecutionList/JobExecutionTable.tsx:27]:
```suggestion
            isSortable={true}
```

**Review by ValentinMumble — APPROVED** [review]

**Review by StevenVAIDIE — APPROVED** [review]

**Pull request merged**